### PR TITLE
Add algorithm

### PR DESCRIPTION
**Pull request by czarska**
https://docs.lib.purdue.edu/cgi/viewcontent.cgi?article=1456&context=cstech&sei-redir=1

### CODE REVIEW

File: exact_string_matching/backward.py [hunk 1]
```diff
@@ -116,3 +116,23 @@ def turbo_boyer_moore(t, w, n, m):
     else:
       i = i + shift
       memory = min(m - shift, match)
+
+def Q(i, k, m, w):
+  return (k <= i and w[(i-k+1):i] == w[(m-k+1):m]) or (k > i and w[1:i] != w[(m-i+1):m])
```
Comment (krzysztof-turowski) [inline exact_string_matching/backward.py:121]: Please put it as an inner function of boyer_moore_apostolico_giancarlo - this way you can remove m and w from the parameters list.

Moreover, is write w[something:m] supposed to omit the last element or should it be rather w[something:]?

**Review by krzysztof-turowski — CHANGES_REQUESTED** [review]
There are still some issues remaining - first one is secondary, but other two are essential:
- "make check" point to a few style problems, please correct them,
- large tests (i.e. "LARGE=1 python3 -B -m unittest test.test_exact_string_matching -v") fail on both random and exhaustive small cases, which effectively points to a bug in your code
- there is missing the pdf/latex description of the algorithm.

File: exact_string_matching/backward.py (hunk 1 above)
Comment (czarska) [reply exact_string_matching/backward.py]: I belive it should omit first and last one

Comment (czarska) [reply exact_string_matching/backward.py]: I will add description of corse, just want to start with algorithm to check if it works

Comment (krzysztof-turowski) [reply exact_string_matching/backward.py]: Thank you very much, everything is working now.

**Review by krzysztof-turowski — APPROVED** [review]

**Pull request merged**